feature-libs/product-configurator/common/components/configure-product/configure-product.component.ts

```diff
@@ -76,6 +77,38 @@ export class ConfigureProductComponent {
     return this.isConfiguratorTypeReadOnly(configuratorType) ? 'false' : 'true';
   }
 
+  /**
+   * Verifies whether a configurator type of a product contains a read-only postfix and
+   * a product is a base product.
+   *
+   * @param product - product
+   * @returns - If the configurator type contains a read-only postfix and
+   * a product is a base product then returns true, otherwise false.
+   */
+  isReadOnlyBaseProduct(product: Product): boolean {
+    return (
+      this.isConfiguratorTypeReadOnly(product.configuratorType) &&
+      this.isBaseProduct(product)
+    );
+  }
+
+  /**
+   * Verifies if a product is a base product.
+   *
+   * In the catalog the product is a base one when baseProduct and code are defined and equal.
+   * In the product detail page a product is a base one when only code is defined.
```

Comment:
the different behavior is really strange.....

i would phrase the comment like this:
```suggestion
   * The `baseProduct` property contains the product code of the base product.
   * So if the baseProduct is undefined or if the provided code equals the code of the product itself,
   * we assume this product to be a base product.
```